CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ else()
 endif()
 
 # Top-level CMake config
-set(CMAKE_CXX_FLAGS "-Wall")
+set(CMAKE_CXX_FLAGS "-Wall -Werror=vla")
```

Comment:
So, what does this do exactly? And what errors does it help prevent?

Reply:
It makes creating a variable-length array a compiler error to prevent any more VLAs sneaking in in the future, e.g.:
```
/usr/local/code/faasm/tests/test/threads/test_levels.cpp:44:20: error: variable length arrays are a C99 feature [-Werror,-Wvla-extension]
    uint8_t buffer[bufferSize];
                   ^~~~~~~~~~
```

Reply:
ah! now i get it. thanks for that.

Reply:
They're easy to miss, as a VLA is created any time you give a non-constant expression as the array size in a `type name[size];` declaration. `std::array<type, size>` won't accept non-constant sizes in the first place and it makes it easy to access the size of the array via `var.size()` so it's also a good replacement.

Reply:
Yes, its good that we can enforce this from the compiler. This reminded me of #199. This PR is not the place to change it, but do you think its worth the re-factor `uint8_t -> std::byte`?

I'd like to progressively chip away on the issues that have been sitting there for months.

Reply:
Yes, it won't impact correctness in any way so it's not high priority but it will make it clearer which variables hold raw memory areas and make it less likely to make mistakes accessing them as 8-bit integers. One downside is that the library apis mostly don't use it so there'll have to be casts at that boundary.